models/quality_measures/intermediate/nqf0041_influenza_immunization/quality_measures__int_nqf0041_denominator.sql

```diff
@@ -0,0 +1,167 @@
+    from {{ ref('quality_measures__stg_core__procedure') }} procedures
+    inner join {{ ref('quality_measures__int_nqf0041__performance_period') }}  as pp
+        on procedure_date between pp.performance_period_begin and  pp.performance_period_end
+    inner join visit_codes
+        on coalesce(procedures.normalized_code,procedures.source_code) = visit_codes.code
+
+)
+
+, claims_encounters as (
+    
+    select 
+          patient_id
+        , coalesce(claim_start_date,claim_end_date) as min_date
+        , coalesce(claim_end_date,claim_start_date) as max_date
+    from {{ ref('quality_measures__stg_medical_claim') }} medical_claim
+    inner join {{ ref('quality_measures__int_nqf0041__performance_period') }}  as pp on
+        coalesce(claim_end_date,claim_start_date)  >=  pp.performance_period_begin
+         and coalesce(claim_start_date,claim_end_date) <=  pp.performance_period_end
+    inner join visit_codes
+        on medical_claim.hcpcs_code= visit_codes.code
+
+)
+
+, all_encounters as (
+
+    select *, 'v' as visit_enc, cast(null as {{ dbt.type_string() }}) as proc_enc, cast(null as {{ dbt.type_string() }}) as claim_enc
+    from visits_encounters
+
+    union all
+
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc, 'p' as proc_enc, cast(null as {{ dbt.type_string() }}) as claim_enc
+    from procedure_encounters
+
+    union all
+    
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc, cast(null as {{ dbt.type_string() }}) as proc_enc, 'c' as claim_enc
+    from claims_encounters
+
+)
+
+, encounters_by_patient as (
+
+    select patient_id, min(min_date) min_date, max(max_date) max_date,
+        concat(concat(
+              coalesce(min(visit_enc),'')
+            , coalesce(min(proc_enc),''))
+            , coalesce(min(claim_enc),'')
+            ) as qualifying_types
+    from all_encounters
+    group by patient_id
+
+)
+
+, patients_with_age as (
+
+    select
+          p.patient_id
+        , min_date
+        , round({{ datediff('birth_date', 'e.max_date', 'hour') }} / 8760.0, 1)  as age_in_one_decimal_point
+        , max_date
+        , qualifying_types
+    from {{ ref('quality_measures__stg_core__patient') }} p
+    inner join encounters_by_patient e
+        on p.patient_id = e.patient_id
+    where p.death_date is null
+
+)
+
+, qualifying_patients as (
+
+    select
+        distinct
+          patients_with_age.patient_id
+        , floor(patients_with_age.age_in_one_decimal_point) as age
+        , pp.performance_period_begin
+        , pp.performance_period_end
+        , pp.measure_id
+        , pp.measure_name
+        , pp.measure_version
+        , 1 as denominator_flag
+    from patients_with_age
+    cross join {{ ref('quality_measures__int_nqf0041__performance_period') }} pp
+    where age_in_one_decimal_point >= 0.5
+
+)
+
+, add_data_types as (
+
+    select
+          cast(patient_id as {{ dbt.type_string() }}) as patient_id
+        , cast(age as integer) as age
```

Comment:
this needs to be set as `float` to support ages below 1.
```suggestion
        , cast(age as float) as age
```

Reply:
this change might be a **BREAKING CHANGE** as on the summary long models, as it unions all the measures information.
also if implemented, this needs to be relayed to the Tuva team

Reply:
Since age field is not used down the lineage, it will not be a Breaking change. Also, float data type is not supported by many sql flavors, so used dbt numeric function. Thank you for the idea.